linkml/cli/main.py

```diff
@@ -0,0 +1,141 @@
+
+# --------------------------------------------------
+# Command groups
+# --------------------------------------------------
+
+
+@click.group()
+def linkml():
+    """
+    LinkML: A flexible linked data modeling language
+    """
+
+
+@linkml.group()
+def generate():
+    """
+    Generate formats from a LinkML schema
+    """
+
+
+@linkml.group(invoke_without_command=True)
+@click.pass_context
+def validate(ctx: click.Context):
+    """
+    Validate a LinkML Schema
+    """
+    if ctx.invoked_subcommand is None:
+        ctx.forward(linkml_validate)
+
+
+@linkml.group()
+def dev():
+    """
+    Helper tools for linkml development
+    """
+
+
+# --------------------------------------------------
+# Add commands to groups
+# --------------------------------------------------
+
+# Top-level linkml commands
+linkml.add_command(linkml_convert, name="convert")
+linkml.add_command(linkml_lint, name="lint")
+linkml.add_command(linkml_sqldb, name="sqldb")
+linkml.add_command(linkml_schema_fixer, name="fix")
+linkml.add_command(linkml_run_examples, name="examples")
+
+# Generators
+generate.add_command(gen_jsonld_context, name="jsonld-context")
+generate.add_command(gen_prefix_map, name="prefix-map")
+generate.add_command(gen_csv, name="csv")
+generate.add_command(gen_graphviz, name="graphviz")
+generate.add_command(gen_golang, name="golang")
+generate.add_command(gen_golr_views, name="golr-views")
+generate.add_command(gen_graphql, name="graphql")
+generate.add_command(gen_java, name="java")
+generate.add_command(gen_jsonld, name="jsonld")
+generate.add_command(gen_json_schema, name="json-schema")
+generate.add_command(gen_markdown, name="markdown")
+generate.add_command(gen_doc, name="doc")
+generate.add_command(gen_namespaces, name="namespaces")
+generate.add_command(gen_owl, name="owl")
+generate.add_command(gen_plantuml, name="plantuml")
+generate.add_command(gen_proto, name="proto")
+generate.add_command(gen_python, name="python")
+generate.add_command(gen_pydantic, name="pydantic")
+generate.add_command(gen_rdf, name="rdf")
+generate.add_command(gen_shex, name="shex")
+generate.add_command(gen_shacl, name="shacl")
+generate.add_command(gen_sparql, name="sparql")
+generate.add_command(gen_typescript, name="typescript")
+generate.add_command(gen_terminusdb, name="terminusdb")
+generate.add_command(gen_yuml, name="yuml")
+generate.add_command(gen_yaml, name="yaml")
+generate.add_command(gen_erdiagram, name="erdiagram")
+generate.add_command(gen_sqla, name="sqla")
+generate.add_command(gen_sqltables, name="sqltables")
+generate.add_command(gen_summary, name="summary")
+generate.add_command(gen_project, name="project")
+generate.add_command(gen_excel, name="excel")
+generate.add_command(gen_sssom, name="sssom")
+generate.add_command(gen_linkml, name="linkml")
+
+# Validators
+validate.add_command(linkml_jsonschema_validate, name="jsonschema")
+validate.add_command(linkml_sparql_validate, name="sparql")
```

Comment:
These are considered legacy at this point. I'd be in favor of not including them and only providing access to the CLI defined in the `linkml.validator` package.

Reply:
Gotcha, didnt know that. Should we put deprecation warnings on those and set a version for removal?

Reply:
Yes, I think so. I can take a crack at following the new (to me, anyway 😁) deprecation process docs for the old `linkml.valiators` CLIs in a separate PR if you just want to drop them from this PR.